bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/Runtime.java

```diff
@@ -54,23 +61,46 @@ public static Runtime getCurrentRuntime() {
-      *
-      * @param object     Object Value.
-      * @param methodName Name of the method.
-      * @param strandName Name for newly creating strand which is used to execute the function pointer. This is optional
-      *                   and can be null.
-      * @param metadata   Meta data of new strand.
-      * @param callback   Callback which will get notify once method execution done.
-      * @param args       Ballerina function arguments.
-      * @return the result of the function invocation
-      */
-     public Object invokeMethodAsync(BObject object, String methodName, String strandName, StrandMetadata metadata,
-                                     Callback callback, Object... args) {
-         Function<?, ?> func = o -> object.call((Strand) (((Object[]) o)[0]), methodName, args);
-         return scheduler.schedule(new Object[1], func, null, callback, strandName, metadata).result;
-     }
+    /**
+     * Invoke Object method asynchronously. This will schedule the function and block the strand.
+     *
+     * @param object     Object Value.
+     * @param methodName Name of the method.
+     * @param strandName Name for newly creating strand which is used to execute the function pointer. This is optional
+     *                   and can be null.
+     * @param metadata   Meta data of new strand.
+     * @param callback   Callback which will get notify once method execution done.
+     * @param args       Ballerina function arguments.
+     * @return the result of the function invocation
+     */
+    public Object invokeMethodAsync(BObject object, String methodName, String strandName, StrandMetadata metadata,
+                                    Callback callback, Object... args) {
+        Function<?, ?> func = o -> object.call((Strand) (((Object[]) o)[0]), methodName, args);
+        boolean isIsolate = isIsolated(object.getType(), methodName);
+        if (isIsolate) {
+            return scheduler.schedule(new Object[1], func, null, callback, strandName, metadata).result;
+        } else {
+            return scheduler.scheduleToObjectGroup(object, new Object[1], func, null, callback, strandName, metadata)
+                    .result;
+        }
+    }
+
+    private boolean isIsolated(ObjectType type, String methodName) {
+        for (MethodType method : type.getMethods()) {
+            if (method.getName().equals(methodName)) {
+                return SymbolFlags.isFlagOn(method.getFlags(), SymbolFlags.ISOLATED);
+            }
+        }
+        if (type.getTag() == SERVICE_TAG) {
+            for (ResourceMethodType method : ((BServiceType) type).getResourceMethods()) {
+                if (method.getName().equals(methodName)) {
+                    return SymbolFlags.isFlagOn(method.getFlags(), SymbolFlags.ISOLATED);
+                }
+            }
+        }
+        assert false : "object type does not contain method : " + methodName;
```

Comment:
I think we do not need this check. We already codegen no such error method if we call non defined object method.

Reply:
it's `assert` so it will get removed by jvm. just here to catch code issues.